catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/ResourceCache.java

```diff
@@ -308,7 +312,7 @@ public Resource getValid(String key, Metacard latestMetacard) {
         // cache directory has had files deleted from it.
         if (cachedResource != null) {
             if (!validateCacheEntry(cachedResource, latestMetacard)) {
-                throw new IllegalArgumentException(
+                throw new CacheException(
                         "Entry found in cache was out-of-date or otherwise invalid.  Will need to be re-cached.  Entry key: "
                                 + key);
             }
```

Comment:
This is basically a cache entry invalidation, which seems like an odd thing to throw an exception for.  I would expect us to log the invalidation and return null.